packages/pwa-kit-runtime/src/utils/logger-factory.js

```diff
@@ -19,6 +19,44 @@ export class PWAKitLogger {
         this.packageName = options.packageName || ''
     }
 
+    /**
+     * Serializes objects for logging, mainly for handling Error objects
+     * @param {*} obj - The object to serialize.
+     * @returns {*} - A serializable version of the object.
+     */
+    #serializeForLogging(obj) {
+        if (obj instanceof Error) {
+            const errorObj = {
+                name: obj.name,
+                message: obj.message,
+                stack: obj.stack,
+                ...(obj.cause ? this.#serializeForLogging(obj.cause) : {})
+            }
+            return errorObj
+        }
+
+        // For arrays, only serialize Error objects within them
+        if (Array.isArray(obj)) {
+            return obj.map((item) =>
+                item instanceof Error ? this.#serializeForLogging(item) : item
+            )
+        }
+
+        // For plain objects, only go one level deep to handle Error properties
+        if (obj && typeof obj === 'object' && obj.constructor === Object) {
+            const serialized = {}
+            for (const [key, value] of Object.entries(obj)) {
+                serialized[key] =
+                    value instanceof Error || Array.isArray(value)
+                        ? this.#serializeForLogging(value)
+                        : value
+            }
```

Comment:
Should we wrap the logic in a try/catch in case anything goes wrong with the serialization?

Reply:
I have simplified the implementation of this method. Please take a look again.

Previously, it was using recursion, which I'm worried that it may run into infinite loop or out-of-memory. And then it tried to be robust and handle several different scenarios. 

Now the implementation has a much more narrow scope: if the given object has some properties that are Error, we will serialize them.